cachi2/core/package_managers/general.py

```diff
@@ -6,9 +6,9 @@
 
 import requests
 
-from cachi2._compat.checksum import hash_file
 from cachi2._compat.errors import InvalidChecksum, NetworkError, UnknownHashAlgorithm
-from cachi2._compat.requests import SAFE_REQUEST_METHODS, get_requests_session
+from cachi2.core.requests import SAFE_REQUEST_METHODS, get_requests_session
```

Comment:
I would prefer to keep only package manager files inside this package (gomod.py, pip.py, npm.py). I feel like the functions here fit better in the `utils.py`. 

The downside is that it could keep growing and become chaotic.

Reply:
My main concern with re-structuring modules was that it would make copying npm, yarn and ruby code even more difficult

But maybe if it's simply moving everything from general -> utils then it would be manageable

Reply:
If you also feel like the functions here look like utils, I think it's worth the trouble of moving them.

Reply:
Hmm, it feels like it might be more complicated than that.
* `ChecksumInfo` class and checksum-related functions could have their own module
* `extract_git_info` could possibly go to scm.py, although it's more pip-specific and also used for ruby
* `pkg_requests_session` should maybe go to http_requests.py? Or directly to pip.py? But then `download_binary_file` would need a different session.

Reply:
Nice, I like where this is heading. :innocent: 
Should we refactor this in a following PR?

Reply:
Yeah that might be better, I feel like this one is large enough already :smile:

But I'm kinda reluctant to make changes which will make copying more package managers harder (_said he, having designed an incompatible error system_). Do you think the trade-off is worth it?

The alternative would be filing a ticket whose priority we can never justify :-/

Reply:
(By right away I meant, in a following PR :grin:)

Reply:
ack, that makes sense